src/components/NavigationStack/NavigationStackItem.tsx

```diff
@@ -0,0 +1,49 @@
+import { memo, ReactElement, useEffect, useMemo, useRef } from 'react';
+import { Animated } from 'react-native';
+import { useNavigation } from './utils';
+
+export type Props = {
+    name: string;
+    children: ReactElement;
+};
+
+const NavigationStackItem = memo(({ name, children }: Props) => {
+    const { currentRoute } = useNavigation();
```

Comment:
make this absolutely dumb.
pull put useNavigation into a separate component that wraps this NavigationStackItem component.

Reply:
updated